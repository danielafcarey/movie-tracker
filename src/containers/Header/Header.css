header {
  background-color: #80808059; }
  header .title {
    display: inline-block;
    width: calc(50% - 40px);
    margin: 0 0 20px;
    text-align: center;
    font-size: 80px; }
  header .nav-bar {
    display: inline-flex;
    justify-content: flex-end;
    align-items: flex-end;
    padding: 0 20px 30px;
    height: 200px;
    width: 50%; }
    header .nav-bar a {
      text-align: right;
      margin: 20px 20px 0;
      font-size: 20px;
      text-decoration: none;
      color: #D46A6A; }
