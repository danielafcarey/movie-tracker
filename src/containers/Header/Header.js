import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { NavLink, Link } from 'react-router-dom';
import { updateCurrentUser, updateFavorites} from '../../actions';

class Header extends Component {
 
  handleClick = () => {
    this.props.updateCurrentUser(null);
    this.props.updateFavorites([]);
  };

  render() {
    return (
      <header>
        <h1 className='title'>MovieTracker and Chill</h1>
        <div className='nav-bar'>
          <NavLink to='/'>Movies</NavLink>
          <NavLink to='/favorites'>Favorites</NavLink>
          <NavLink to='/signup'>Sign Up</NavLink>
          <NavLink to='/login'>Log In</NavLink>
          <Link to='/' onClick={this.handleClick}>Log Out</Link>
        </div>
      </header>
    );
  };

};

const mapDispatchToProps = (dispatch) => ({
  updateCurrentUser: (id) => dispatch(updateCurrentUser(id)),
  updateFavorites: (favorites) => dispatch(updateFavorites(favorites))
});

Header.propTypes = {
  updateCurrentUser: PropTypes.func,
  updateFavorites: PropTypes.func
}

export {
  Header,
  mapDispatchToProps
};

export default Header;
