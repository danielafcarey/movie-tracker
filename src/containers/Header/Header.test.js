import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapDispatchToProps } from './Header';

describe('Header', () => {

  let wrapper;
  let mockUpdateCurrentUser;
  let mockUpdateFavorites;

  beforeEach(() => {
    mockUpdateCurrentUser = jest.fn();
    mockUpdateFavorites = jest.fn();
    wrapper = shallow(<Header
        updateCurrentUser={mockUpdateCurrentUser}
        updateFavorites={mockUpdateFavorites}
       />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('handleClick', () => {

    it('calls props.updateCurrentUser with correct argument', () => {
      const wrapperInst = wrapper.instance();
      wrapperInst.handleClick();
      expect(wrapperInst.props.updateCurrentUser).toHaveBeenCalledWith(null)
    })

    it('calls props.updateFavorites with correct argument', () => {
      const wrapperInst = wrapper.instance();
      wrapperInst.handleClick();
      expect(wrapperInst.props.updateFavorites).toHaveBeenCalledWith([])
    })  
  })

  describe('mapDispatchToProps', () => {

    describe('updateCurrentUser', () => {
      it('returns an object with an updateCurrentUser function', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);

        expect(typeof result.updateCurrentUser).toEqual('function');
      });

      it('calls dispatch with the correct argument', () => {
        const dispatch = jest.fn();
        const mappedProps = mapDispatchToProps(dispatch)
        const mockAction = {
          type: 'UPDATE_CURRENT_USER',
          id: null
        }

        mappedProps.updateCurrentUser(mockAction.id)

        expect(dispatch).toHaveBeenCalledWith(mockAction)
      })
    })

    describe('updateFavorites', () => {
      it('returns an object with an updateFavorites function', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);

        expect(typeof result.updateFavorites).toEqual('function');
      });

      it('calls dispatch with the correct argument', () => {

        const dispatch = jest.fn();
        const mappedProps = mapDispatchToProps(dispatch)
        const mockAction = {
          type: 'UPDATE_FAVORITES',
          favorites: []
        }

        mappedProps.updateFavorites([])

        expect(dispatch).toHaveBeenCalledWith(mockAction)
      })
    })

  })
})
