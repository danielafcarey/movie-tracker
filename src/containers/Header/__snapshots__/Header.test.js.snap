// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/"
>
          Movies
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/favorites"
>
          Favorites
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/login"
>
          LogIn
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/signup"
>
          SignUp
</NavLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Movies",
          "to": "/",
        },
        "ref": null,
        "rendered": "Movies",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Favorites",
          "to": "/favorites",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "LogIn",
          "to": "/login",
        },
        "ref": null,
        "rendered": "LogIn",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "SignUp",
          "to": "/signup",
        },
        "ref": null,
        "rendered": "SignUp",
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/"
>
            Movies
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/favorites"
>
            Favorites
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/login"
>
            LogIn
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/signup"
>
            SignUp
</NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Movies",
            "to": "/",
          },
          "ref": null,
          "rendered": "Movies",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Favorites",
            "to": "/favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "LogIn",
            "to": "/login",
          },
          "ref": null,
          "rendered": "LogIn",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "SignUp",
            "to": "/signup",
          },
          "ref": null,
          "rendered": "SignUp",
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
